### ISSUE TITLE
Roles

### ISSUE THREAD

**Issue by corlaez** [corlaez/asistencia#6]
Venue (role venue crud)
Leader (role leader crud)
Group (role group crud) owned by leader
- Group transfer to a new leader
- Add a participant to group
- Create attendance records
- Role to record attendance (for a temporal leader)
- Role to own all groups (for admins)

Participant (role participant crud)

Attendance Record

- Persists date time, groupOwnerLeaderId, creatorLeaderId, participant, groupId, groupName, VenueId, VenueName. (Also persist names of leaders and participants).
- Consider to record absence or late explicitly as well

**Comment by corlaez** [comment]
Conversation with Jessica: The user talked about another user that would follow up with leaders to check if the leaders are registering attendance when they should.
- Add a role that allows reading all attendance records.
- Scheduling is another alternative but we may not want to buy into that.

**Comment by JessicaMP** [comment]
Scheduling is another alternative but in my opinion we should only focus on the attendance requirements.
Regarding the roles, these 3 are the ones that I am clear at the moment:

- Administrator (Access to everything)
- Leader (Attendance and monitoring data of their respective group)
- Auditor (Read-only permissions)

**Comment by corlaez** [comment]
Ok yeah I agree with that.
Administrator (Access to everything)
Leader (Create Groups, Edit Owned Groups, Create Attendance Records on their Group Members. Additionally, as nice to haves: Offer Ownership Transfer, Accept Ownership Transfer, add/remove a secondary leader(s) to the group.)
Auditor (Read-only permissions on attendance for all groups)

attendance records should be created for all the members of a group wether they assisted or not.

**Comment by corlaez** [comment]
Q: do we also want to offer a “Late” option?
I think it wouldn’t be hard to add and a lot of users could benefit from such an option

**Comment by JessicaMP** [comment]
I believe Offer Ownership Transfer, Accept Ownership Transfer, add/remove a secondary leader(s) to the group this actions should only role Administrator

**Comment by corlaez** [comment]
Admin will have that ability for sure, after all he can do everything.
However, I believe that it makes sense that any owner of a group can offer the ownership to another leader. Then that leader can accept or reject the responsibility to be the new owner.
Adding and removing secondary leaders may need to go through am offer/accept flow to be honest. And same thing applies as in the owner transfer.
The difference is that at any point the leader could delete the secondary leader, without going through an offer/accept flow.
All of these are nice to haves and not core of the system, but that is my rationale for them being not only for the admin role

**Comment by JessicaMP** [comment]
Yes, I agree. I belive this functions are appropriate for admin and leader
